x-pack/plugins/maps/public/routing/bootstrap/services/saved_gis_map.ts

```diff
@@ -18,73 +23,80 @@ import {
 } from '../../../selectors/map_selectors';
 import { getIsLayerTOCOpen, getOpenTOCDetails } from '../../../selectors/ui_selectors';
 import { copyPersistentState } from '../../../reducers/util';
+// @ts-expect-error
 import { extractReferences, injectReferences } from '../../../../common/migrations/references';
 import { getExistingMapPath, MAP_SAVED_OBJECT_TYPE } from '../../../../common/constants';
+// @ts-expect-error
 import { getStore } from '../../store_operations';
+import { MapStoreState } from '../../../reducers/store';
+import { LayerDescriptor } from '../../../../common/descriptor_types';
+
+export interface ISavedGisMap extends SavedObject {
+  layerListJSON?: string;
+  mapStateJSON?: string;
+  uiStateJSON?: string;
+  getLayerList(): LayerDescriptor[];
+  syncWithStore(): void;
+}
 
-export function createSavedGisMapClass(services) {
+export function createSavedGisMapClass(services: SavedObjectKibanaServices) {
   const SavedObjectClass = createSavedObjectClass(services);
 
-  class SavedGisMap extends SavedObjectClass {
-    static type = MAP_SAVED_OBJECT_TYPE;
+  class SavedGisMap extends SavedObjectClass implements ISavedGisMap {
+    public static type = MAP_SAVED_OBJECT_TYPE;
 
     // Mappings are used to place object properties into saved object _source
-    static mapping = {
+    public static mapping = {
       title: 'text',
       description: 'text',
       mapStateJSON: 'text',
       layerListJSON: 'text',
       uiStateJSON: 'text',
     };
-    static fieldOrder = ['title', 'description'];
-    static searchSource = false;
+    public static fieldOrder = ['title', 'description'];
+    public static searchSource = false;
 
-    constructor(id) {
+    public showInRecentlyAccessed = true;
+    public layerListJSON?: string;
+    public mapStateJSON?: string;
+    public uiStateJSON?: string;
+
+    constructor(id: string) {
       super({
         type: SavedGisMap.type,
         mapping: SavedGisMap.mapping,
         searchSource: SavedGisMap.searchSource,
         extractReferences,
-        injectReferences: (savedObject, references) => {
+        injectReferences: (savedObject: ISavedGisMap, references: SavedObjectReference[]) => {
           const { attributes } = injectReferences({
             attributes: { layerListJSON: savedObject.layerListJSON },
             references,
           });
 
           savedObject.layerListJSON = attributes.layerListJSON;
-
-          const indexPatternIds = references
-            .filter((reference) => {
-              return reference.type === 'index-pattern';
-            })
-            .map((reference) => {
-              return reference.id;
-            });
-          savedObject.indexPatternIds = _.uniq(indexPatternIds);
```

Comment:
`savedObject.indexPatternIds` was added when MapEmbeddable was first introduced in https://github.com/elastic/kibana/pull/31473/files#diff-1bb10565f2889677ad2ded8210390726R50. At the time MapEmbeddableFactory used `savedObject.indexPatternIds` to obtain index patterns used by the map

https://github.com/elastic/kibana/pull/35542 changed the way MapEmbeddableFactory obtained indexPatternIds to pulling them from store, https://github.com/elastic/kibana/commit/acb78a60ba48304c812888965f39909a6c55bf16#diff-e347d1b056afd109215cc604d63d4aefL47. However, that PR failed to clean up setting indexPatternIds on savedMap.

Since savedMap.indexPatternIds is no longer used, it is safe to remove this code